### PR TITLE
fix(esl-toggleable): doesn't update after changing the activator

### LINKED ISSUE #1677: [🐛 in esl-toggleable]: doesn't update after changing the activator for a component in the open state
**To Reproduce**
There is one `esl-popup` (the descendant of `esl-toggleable`) and two `esl-triggers` that activate the popup. Triggers activate the popup on hover. The popup closes with a certain delay. The delay is sufficient to move the mouse cursor from one trigger to the other before the popup closes. 
If you move the cursor to the 1st trigger, the popup opens. The position of the popup will be relative to the 1st trigger depending on its parameters.
If you move the cursor quickly to the 2nd trigger, move it so fast that the popup does not close.

**Current behavior**
The popup will not close. But it will still be positioned relative to the 1st trigger.

**Expected behavior**
The popup must remain open and its position must be relative to the 2nd trigger.

### PR DESCRIPTION
**Pull request by dshovchko**
Fixed incorrect behavior of toggleable when the activator changes in the open state.

Closes #1677

### CODE REVIEW

**Review by ala-n — APPROVED** [review]
Cosmetic: posibly some shorter name for the new hook. Let's discuss on Monday meeting.

**Review by fshovchko — APPROVED** [review]

**Review by Natalie-Smirnova — APPROVED** [review]

**Review by HenadzV — APPROVED** [review]

File: src/modules/esl-toggleable/core/esl-toggleable.ts [hunk 1]
```diff
@@ -196,9 +196,16 @@ export class ESLToggleable extends ESLBaseElement {
     return this;
   }
 
+  /** @returns whether the hide task should be executed for the open state during the show task executing */
+  protected shouldHideWhenOpen(params: ESLToggleableActionParams): boolean {
```
Comment (ala-n) [inline src/modules/esl-toggleable/core/esl-toggleable.ts:200]: `shouldHideIfOpen`

**Review by fshovchko — APPROVED** [review]

**Review by fshovchko — APPROVED** [review]

File: src/modules/esl-toggleable/core/esl-toggleable.ts [hunk 2]
```diff
@@ -196,9 +196,17 @@ export class ESLToggleable extends ESLBaseElement {
...
+   * @returns whether the show task should be run to the end.
+   */
+  protected onParamsUpdate(params: ESLToggleableActionParams): boolean | void {
+    this.activator = params.activator;
+  }
+
   /** Actual show task to execute by toggleable task manger ({@link DelayedTask} out of the box) */
   protected showTask(params: ESLToggleableActionParams): void {
-    if (!params.force && this.open) return;
+    if (!params.force && this.open && !this.onParamsUpdate(params)) return;
```
Comment (ala-n) [inline src/modules/esl-toggleable/core/esl-toggleable.ts:209]: Not sure about the case with `force=true` in terms of whether should we skip the onParmsUpdate execution

**Review by ala-n — APPROVED** [review]

**Review by julia-murashko — APPROVED** [review]

**Review by NastaLeo — APPROVED** [review]

**Review by HenadzV — APPROVED** [review]

**Comment by ala-n** [comment]
Accepted for release 4.7.0

**Review by abarmina — APPROVED** [review]

**Pull request merged**